docs/source/concepts/ledger-model/ledger-exceptions.rst

```diff
@@ -122,6 +122,84 @@ And the third:
 
 As you can see, in each of these continuities, no contract was consumed twice.
 
+Transaction Normalization
++++++++++++++++++++++++++
+
+The same "before-after" relation can be represented in more than one way using
+rollback nodes. For example, the following three transactions have the same
+"before-after" relation among their ledger actions (A and B):
+
+.. https://lucid.app/lucidchart/3aa5922f-ec30-4896-8bbc-56703549c7e5/edit
+.. image:: ./images/exception-normalization-1.svg
+   :align: center
+   :width: 80%
+
+Because of this, these three transactions are equivalent.
+More generally, two transactions are equivalent if:
+
+- The transactions are the same when you ignore all rollback nodes. That is,
+  if you remove every rollback node and absorb its children into its parent,
+  then two transactions are the same. Equivalently, the transactions have
+  the same ledger actions with the same prefix order and subaction relation.
```

Comment:
do you mean prefix order or the same preorder traversal?

Reply:
Ah, preorder traversal makes more sense :-)